yt/frontends/gadget/data_structures.py

```diff
@@ -666,24 +666,12 @@ def _set_owls_eagle_units(self):
 
     @classmethod
     def _is_valid(cls, filename, *args, **kwargs):
-        need_groups = ["Header"]
-        veto_groups = ["FOF", "Group", "Subhalo"]
-        valid = True
+        # need_groups = ["Header"]
+        # veto_groups = ["FOF", "Group", "Subhalo"]
         try:
-            fh = h5py.File(filename, mode="r")
-            valid = all(ng in fh["/"] for ng in need_groups) and not any(
-                vg in fh["/"] for vg in veto_groups
-            )
-            fh.close()
+            with h5py.File(filename, mode="r") as fh:
+                # valid = all(ng in fh["/"] for ng in need_groups)
```

Comment:
On the main branch, the first try block has no lasting effect on the value of `valid`. In my refactor, I tried making it relevant and it broke a lot of tests, so now I'm commenting out the broken bits and I see no path for them but removal unless someone with more knowledge of the frontend could jump in and tell me how it should be fixed. From the git blame, I _think_ @brittonsmith might be able to play that role but it's not really clear as this function has a lot of contributors.

Reply:
If I am remembering correclty, this is trying to discern between a couple different very-mildly-different formats, so it's probably a bit tricky.

Reply:
Well the road was indubitably bumpy, but I think I got the PR to a stable state with not YTAmbiguousDataFormat or YTUnidentifiedDataFormat so I'm a lot more confident that this is correct

Reply:
> indubitably 

😂 thank you for using this amazing word in a PR discussion

Reply:
Tbh I think it is slightly more common (and less ridiculous) in French, but I stand by it :)